code/modules/mob/living/carbon/xenomorph/castes/slime/abilities_slime.dm

```diff
@@ -0,0 +1,480 @@
+	if(xeno_owner.issamexenohive(living_target))
+		living_target.throw_at(xeno_owner, get_dist(living_target, xeno_owner) - 1, 1, xeno_owner, FALSE)
+		living_target.Knockdown(SLIME_EXTEND_TENDRIL_EFFECT_DURATION / 2)
+		return
+	var/distance = get_dist(living_target, xeno_owner)
+	var/damage = xeno_owner.xeno_caste.melee_damage
+	var/effect_duration = SLIME_SPREAD_TENDRILS_EFFECT_DURATION
+	if(distance > 1)
+		living_target.throw_at(xeno_owner, SLIME_EXTEND_TENDRIL_PULL_RANGE, 1, xeno_owner, FALSE)
+		if(distance >= SLIME_EXTEND_TENDRIL_RANGE)
+			damage = damage * SLIME_EXTEND_TENDRIL_CRITICAL_MULTIPLIER
+			effect_duration = effect_duration * SLIME_EXTEND_TENDRIL_CRITICAL_MULTIPLIER
+	new /obj/effect/temp_visual/tendril_blood(get_turf(living_target))
+	living_target.apply_damage(damage, BURN, blocked = ACID)
+	living_target.do_jitter_animation(1 SECONDS)
+	living_target.Stun(effect_duration)
+	living_target.adjust_stagger(effect_duration)
+	living_target.emote("scream")
+	shake_camera(living_target, 2, 1)
+	if(HAS_TRAIT(living_target, TRAIT_INTOXICATION_IMMUNE))
+		return
+	var/applied_stacks = SLIME_EXTEND_TENDRIL_INTOXICATION_STACKS + xeno_owner.xeno_caste.additional_stacks
+	if(living_target.has_status_effect(STATUS_EFFECT_INTOXICATED))
+		var/datum/status_effect/stacking/intoxicated/debuff = living_target.has_status_effect(STATUS_EFFECT_INTOXICATED)
+		debuff.add_stacks(applied_stacks)
+	else
+		living_target.apply_status_effect(STATUS_EFFECT_INTOXICATED, applied_stacks)
+
+/// Handles the beam used in representation of the Slime's tendril.
+/// Has a "mode" var. If set to true, it will delete the beam instead of creating one.
+/datum/action/xeno_action/activable/extend_tendril/proc/handle_beam(mode, atom/target_atom)
+	if(!mode)
+		tendril_beam = owner.beam(target_atom, icon_state = "tentacle", beam_type = /obj/effect/ebeam/slime_tendril)
+		addtimer(CALLBACK(src, PROC_REF(handle_beam), TRUE, target_atom), 0.5 SECONDS)
+		return
+	QDEL_NULL(tendril_beam)
+
+// ***************************************
+// *********** Toxic Burst
+// ***************************************
+/datum/action/xeno_action/toxic_burst
+	name = "Toxic Burst"
+	ability_name = "Toxic Burst"
+	action_icon_state = "toxic_burst"
+	desc = "Attack a victim you're attached to, consuming the Intoxication debuff and dealing an amount of damage proportional to the amount of stacks."
+	cooldown_timer = 60 SECONDS
+	plasma_cost = 50
+	use_state_flags = XACT_USE_BUCKLED
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_SLIME_TOXIC_BURST,
+	)
+	/// References the Pounce action and its vars.
+	var/datum/action/xeno_action/activable/slime_pounce/slime_pounce_action
+
+/datum/action/xeno_action/toxic_burst/give_action(mob/living/L)
+	. = ..()
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	slime_pounce_action = xeno_owner.actions_by_path[/datum/action/xeno_action/activable/slime_pounce]
```

Comment:
Not mandatory but i'd personally make this be got on demand in case it gets deleted or so